server/src/daemon/service_manager.rs

```diff
@@ -0,0 +1,27 @@
+use std::collections::HashMap;
+
+use tokio::sync::RwLock;
+use zbus::zvariant::{ObjectPath, OwnedObjectPath};
+
+use super::session::Session;
+
+#[derive(Debug)]
+pub struct ServiceManager {
+    sessions: RwLock<HashMap<OwnedObjectPath, Session>>,
+}
+
+impl ServiceManager {
+    pub fn new() -> Self {
+        Self {
+            sessions: RwLock::new(HashMap::new()),
+        }
+    }
+
+    pub async fn session(&self, path: ObjectPath<'_>) -> Option<&Session> {
```

Comment:
> Don't add a getter that is not used, yet. You would just need an API to get a Option<&Session> from a session path

done.

How can I fix the following error,
```
error[E0515]: cannot return value referencing temporary value
  --> server/src/daemon/service_manager.rs:21:9
   |
21 |         self.sessions.read().await.get(&path.into())
   |         --------------------------^^^^^^^^^^^^^^^^^^
   |         |
   |         returns a value referencing data owned by the current function
   |         temporary value created here
```

Reply:
Not if you wrap the session key in Arc<Key>

Reply:
I have updated  `client_public_key: Arc<Option<Key>>`. So, cloning a `Session` works now  But, I'm still getting the following error at line: [21](https://github.com/bilelmoussaoui/oo7/blob/95ebf467092569d14dc9d1cb9d1909822bf87adc/server/src/daemon/service_manager.rs#L21),
`returns a value referencing data owned by the current function`.